### PR TITLE
Properly handle timestamp prefixing of unkown window types.

### PR DESCRIPTION
**Pull request by robertwb**
This was exposed by #28972 when the set of "known" coders was inadvertently reduced.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Abacn

**Comment by shunping** [comment]
Gentle ping on the review request. @Abacn

File: sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py [hunk 1]
```diff
@@ -538,6 +570,7 @@ def maybe_length_prefixed_and_safe_coder(self, coder_id):
                 spec=coder.spec, component_coder_ids=safe_component_ids))
       return new_coder_id, safe_coder_id
     else:
+      # A completely unkown coder. Wrap the entire thing in a length prefix.
```
Comment (Abacn) [inline sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py:573]: I see, the bytes encountered in XVR_Direct test was due to here

**Review by Abacn — APPROVED** [review]
tested `test_windowing_before_sql` locally and passed

For curiosity, what caused the window being "unknown"  for that specific test?

And after this change, it seems the window is still "unknown" but the timestamp value is accessible, or I understood it incorrectly?

**Comment by robertwb** [comment]
> tested `test_windowing_before_sql` locally and passed
> 
> For curiosity, what caused the window being "unknown" for that specific test?

Due to the environment nesting it didn't realize that "interval window" type was understood by all SDKs, so fell back to the generic/unknown wrapping. 

> And after this change, it seems the window is still "unknown" but the timestamp value is accessible, or I understood it incorrectly?

Yep, that's exactly it.

**Comment by Abacn** [comment]
It seems this PR breaks Samza and Spark VR PostCommit, https://github.com/apache/beam/actions/workflows/beam_PostCommit_Python_ValidatesRunner_Samza.yml?query=event%3Aschedule and https://github.com/apache/beam/actions/workflows/beam_PostCommit_Python_ValidatesRunner_Spark.yml?query=event%3Aschedule

though Flink and Dataflow VR PostCommit remains green

Error message

```
test_custom_window_type (apache_beam.runners.portability.spark_runner_test.SparkRunnerTest) failed

E       RuntimeError: Pipeline test_custom_window_type_1710372598.7745032_87e88bc1-0f78-497a-b23a-02d9a534d8ac failed in state FAILED: java.lang.IllegalArgumentException: Unknown or unsupported WindowFn: beam:window_fn:pickled_python:v1

```

**Pull request merged**